core/src/main/java/org/elasticsearch/action/update/UpdateRequestBuilder.java

```diff
@@ -279,23 +330,23 @@ public UpdateRequestBuilder setUpsert(Object... source) {
         return this;
     }
 
-    public UpdateRequestBuilder setSource(XContentBuilder source) throws Exception {
-        request.source(source);
+    public UpdateRequestBuilder fromXContent(XContentBuilder source) throws Exception {
```

Comment:
I have no idea ;) I'd prefer keep it that way for now and remove it later in another PR ?